pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -581,9 +583,22 @@ func (pl *dynamicResources) PreScore(ctx context.Context, cs *framework.CycleSta
 }
 
 func haveAllNodes(nodeNames []string, nodes []*v1.Node) bool {
-	for _, node := range nodes {
-		if !haveNode(nodeNames, node.Name) {
-			return false
+	if len(nodeNames) < len(nodes) {
+		return false
+	}
+	// The time complexity is O(n)
```

Comment:
Let's use `HasAll` directly here, and I think we can also have the  benchmark test if we do want to prove this helps improve the performance. And maybe you can provide the benchmark result with nodes number to 10, 50, 500, which helps to judge whether this is the right direction. WDYT?

Reply:
Or maybe `HasAll` should be more general by introducing this to package sets.

Reply:
`HasAll` would requre to convert list of nodes to the list of node names as far as I understand. This may impact performance.

Reply:
Oh, I didn't notice that. Then we can't use HasAll here.

Reply:
@kerthcet Benchmark test and result are given.

before:
```
BenchmarkHaveAllNodes/have-nodes-10-16         	 5319110	       214.5 ns/op	       0 B/op	       0 allocs/op
BenchmarkHaveAllNodes/have-nodes-50-16         	  336820	      3602 ns/op	       0 B/op	       0 allocs/op
BenchmarkHaveAllNodes/have-nodes-100-16        	   81951	     14545 ns/op	       0 B/op	       0 allocs/op
BenchmarkHaveAllNodes/have-nodes-500-16        	    4078	    293049 ns/op	       0 B/op	       0 allocs/op
```
after:
```
BenchmarkHaveAllNodes/have-nodes-10-16         	 5723989	       207.7 ns/op	       0 B/op	       0 allocs/op
BenchmarkHaveAllNodes/have-nodes-50-16         	  317394	      3879 ns/op	       0 B/op	       0 allocs/op
BenchmarkHaveAllNodes/have-nodes-100-16        	  158844	      7405 ns/op	    2888 B/op	       3 allocs/op
BenchmarkHaveAllNodes/have-nodes-500-16        	   34837	     34149 ns/op	   20552 B/op	       3 allocs/op
```